test/16_holograph_treasury_tests.ts

```diff
@@ -1,80 +1,257 @@
+    holographTreasury = (await holographTreasuryFactory.deploy()) as HolographTreasury;
+    await holographTreasury.deployed();
+
+    const mockExternalCallFactory = await ethers.getContractFactory('MockExternalCall');
+    mockExternalCall = (await mockExternalCallFactory.deploy()) as MockExternalCall;
+    await mockExternalCall.deployed();
+  });
+
+  describe('constructor', async function () {});
+
+  describe('init()', async function () {
+    it('should successfully be initialized once', async () => {
+      const tx = await holographTreasury.connect(deployer).init(initPayload);
+      await tx.wait();
+
+      expect(await holographTreasury.getHolograph()).to.equal(holographMock);
+      expect(await holographTreasury.getOperator()).to.equal(operatorMock);
+      expect(await holographTreasury.getRegistry()).to.equal(registryMock);
+      expect(await holographTreasury.getBridge()).to.equal(bridgeMock);
+    }); // Validate hardcoded values are correct
+
+    it('should fail if already initialized', async () => {
+      const tx = await holographTreasury.connect(deployer).init(initPayload);
+      await tx.wait();
+      await expect(holographTreasury.connect(deployer).init(initPayload)).to.be.revertedWith(
+        'HOLOGRAPH: already initialized'
+      );
+    });
+
+    it('Should allow external contract to call fn', async () => {
+      let ABI = ['function init(bytes memory initPayload)'];
+      let iface = new ethers.utils.Interface(ABI);
+      let encodedFunctionData = iface.encodeFunctionData('init', [initPayload]);
+
+      await expect(mockExternalCall.connect(deployer).callExternalFn(holographTreasury.address, encodedFunctionData)).to
+        .not.be.reverted;
+
+      expect(await holographTreasury.getHolograph()).to.equal(holographMock);
+      expect(await holographTreasury.getOperator()).to.equal(operatorMock);
+      expect(await holographTreasury.getRegistry()).to.equal(registryMock);
+      expect(await holographTreasury.getBridge()).to.equal(bridgeMock);
+    });
+
+    it('should fail to allow inherited contract to call fn', async () => {});
+  });
+
+  describe('After initialized', () => {
+    beforeEach(async () => {
+      const tx = await holographTreasury.connect(deployer).init(initPayload);
+      await tx.wait();
+    });
+
+    function testIfIsPrivate(fnAbi: string, fnName: string, args: any[] = []) {
+      it('is private function', async () => {
+        let iface = new ethers.utils.Interface([fnAbi]);
+        let encodedFunctionData = iface.encodeFunctionData(fnName, args);
+
+        await expect(
+          ethers.provider.call({
+            to: holographTreasury.address,
+            data: encodedFunctionData,
+          })
+        ).to.be.reverted;
+      });
+    }
+
+    describe(`_bridge()`, async function () {
+      // it('should successfully be initialized once')
+      testIfIsPrivate('function _bridge() view returns (address bridge)', '_bridge');
```

Comment:
This seems to be passing for public functions as well

Reply:
Damn, how do you propose to solve that? How can we test that?

Reply:
What do you mean?

Reply:
I used this same helper function for my test and seems to be passing when I provide a public one

Reply:
@ackvf @feliperodriguess  I have my snippet to test if the function is private. Granted it is a bit silly.

```
    it('is private function', async function () {
      expect(typeof holographOperator.connect(deployer)._randomBlockHash).to.equal('undefined')
      expect(holographOperator.connect(deployer)).to.not.have.keys('_randomBlockHash')
    });
  ```

Reply:
I wonder how this works? Does this test on the javascript level or does it do an actual blockchain call?

Reply:
@ackvf it works by a negative. If hardhat reads the ABI and it creates the function, then we know that the function is no longer private. Since it is private, hardhat wont make the function and add it to the contract object.

Reply:
> I used this same helper function for my test and seems to be passing when I provide a public one

You can't put this function inside another `it`, otherwise it not actually executed and the test seems to pass

Reply:
yep, you're right. All good @BravoNatalie ✅